python/dgl/frame.py

```diff
@@ -548,7 +559,14 @@ def record_stream(self, stream):
         """
         if F.get_preferred_backend() != "pytorch":
             raise DGLError("record_stream only supports the PyTorch backend.")
-        self.data.record_stream(stream)
+        # Only record streams for materialized tensors otherwise just record for the index
+        if self.index is None:
+            self.data.record_stream(stream)
+        elif (
+            isinstance(self.index, _LazyIndex)
+            or F.context(self.index) != F.cpu()
+        ):
+            self.index.record_stream(stream)
```

Comment:
Should we be calling `self.storage.record_stream(stream)` whenever `index` exists?

Reply:
How about
```python
        if (
            isinstance(self.index, _LazyIndex)
            or F.context(self.index) != F.cpu()
        ):
            self.index.record_stream(stream)
        if F.context(self.storage) != F.cpu():
            self.storage.record_stream(stream)
```

Reply:
@chang-l How do you like this modification?

Reply:
@yaox12 @chang-l Sounds good! I added a commit with recording on storage. I think if all tests pass, it should be ready to merge.